### PR TITLE
Added a ledger chunk class to manage the Transactions in a single chunk

### PR DESCRIPTION
**Pull request by msftsettiy**
This PR contains changes to split the Ledger class into two. The Ledger class is responsible for iterating over the ledger chunks and the Ledgerchunk class is responsible for iterating over the Transactions in a chunk.

### CODE REVIEW

File: python/ccf/ledger.py [hunk 1]
```diff
@@ -232,21 +232,46 @@ def __next__(self):
             raise StopIteration() from e
 
 
+class Ledgerchunk:
```
Comment (achamayou) [inline python/ccf/ledger.py:235]:
```suggestion
class LedgerChunk:
```

https://www.python.org/dev/peps/pep-0008/#type-variable-names

File: python/ccf/ledger.py [hunk 2]
```diff
@@ -232,21 +232,46 @@ def __next__(self):
...
+    def __init__(self, name: str):
+
+        self._current_tx = Transaction(name)
+        self._filename = name
+
+    def __next__(self) -> Transaction:
+        try:
+            return next(self._current_tx)
+        except StopIteration:
+            LOG.debug(f"\nCompleted verifying Txns in {self._filename}.")
```
Comment (achamayou) [inline python/ccf/ledger.py:254]:
```suggestion
            LOG.debug(f"Completed verifying Txns in {self._filename}.")
```

File: python/ccf/ledger.py [hunk 3]
```diff
@@ -232,21 +232,46 @@ def __next__(self):
...
+            raise
+    
+    def __iter__(self):
+        return self
+
 class Ledger:
     """
-    Class used to parse and iterate over all :py:class:`ccf.ledger.Transaction` stored in a CCF ledger.
+    Class used to iterate over all :py:class:`ccf.ledger.Ledgerchunk` stored in a CCF ledger folder.
```
Comment (achamayou) [inline python/ccf/ledger.py:262]:
```suggestion
    Class used to iterate over all :py:class:`ccf.ledger.LedgerChunk` stored in a CCF ledger folder.
```

File: python/ccf/ledger.py [hunk 4]
```diff
@@ -232,21 +232,46 @@ def __next__(self):
...
 
     :param str name: Ledger directory for a single CCF node.
     """
 
     _filenames: list
     _fileindex: int
-    _current_tx: Transaction
+    _current_chunk: Ledgerchunk
```
Comment (achamayou) [inline python/ccf/ledger.py:269]:
```suggestion
    _current_chunk: LedgerChunk
```

File: python/ccf/ledger.py [hunk 5]
```diff
@@ -264,18 +289,13 @@ def __init__(self, directory: str):
...
-        try:
-            return next(self._current_tx)
-        except StopIteration:
-            self._fileindex += 1
-            if len(self._filenames) > self._fileindex:
-                self._current_tx = Transaction(self._filenames[self._fileindex])
-                return next(self._current_tx)
-            else:
-                raise
+    def __next__(self) -> Ledgerchunk:
```
Comment (achamayou) [inline python/ccf/ledger.py:292]:
```suggestion
    def __next__(self) -> LedgerChunk:
```

File: python/ccf/ledger.py [hunk 6]
```diff
@@ -264,18 +289,13 @@ def __init__(self, directory: str):
...
+        self._fileindex += 1
+        if len(self._filenames) > self._fileindex:
+            self._current_chunk = Ledgerchunk(self._filenames[self._fileindex])
```
Comment (achamayou) [inline python/ccf/ledger.py:295]:
```suggestion
            self._current_chunk = LedgerChunk(self._filenames[self._fileindex])
```

File: python/ccf/ledger.py [hunk 7]
```diff
@@ -264,18 +289,13 @@ def __init__(self, directory: str):
...
+            return self._current_chunk
+        else:
+            raise StopIteration
```
Comment (achamayou) [inline python/ccf/ledger.py:298]:
```suggestion
            raise
```

**Comment by achamayou** [comment]
Thanks @msftsettiy, that change looks good!

**Review by achamayou — APPROVED** [review]

File: python/ccf/ledger.py (hunk 1 above)
Comment (jumaffre) [inline python/ccf/ledger.py:235]: Could you also update `doc/operators/ledger_python.rst` as documentation is automatically created from these classes?

**Comment by jumaffre** [comment]
@msftsettiy Changes look good - thanks! 

Could you format `ledger.py` with `black`? Also, the `governance_history_test` seems to be failing: 

```
38: Traceback (most recent call last):
38:   File "/mnt/build/2/s/tests/governance_history.py", line 124, in <module>
38:     run(args)
38:   File "/mnt/build/2/s/tests/governance_history.py", line 72, in run
38:     ) = count_governance_operations(ledger)
38:   File "/mnt/build/2/s/tests/governance_history.py", line 23, in count_governance_operations
38:     tables = tr.get_public_domain().get_tables()
38: AttributeError: 'Ledgerchunk' object has no attribute 'get_public_domain'
29/51 Test #38: governance_history_test ..........***Failed   11.66 sec
```

It is the only test we have that uses the `ledger.py` file. The fix should be pretty trivial but please reach out if you need a hand.

File: python/ccf/ledger.py (hunk 1 above)
Comment (msftsettiy) [reply python/ccf/ledger.py:235]: Fixed it

Comment (msftsettiy) [reply python/ccf/ledger.py:235]: Sure

File: python/ccf/ledger.py (hunk 7 above)
Comment (msftsettiy) [reply python/ccf/ledger.py:298]: @achamayou, got a runtime error when I removed the StopIteration "RuntimeError: No active exception to reraise".

Comment (achamayou) [reply python/ccf/ledger.py:298]: My bad, you're absolutely right, I was looking at the left of the diff and failed to spot this isn't an except path anymore. Sorry.

**Comment by achamayou** [comment]
@msftsettiy could you run './scripts/ci-checks.sh -f' to apply canonical formatting to your changes please? The PR is otherwise ready to merge.

**Pull request merged**